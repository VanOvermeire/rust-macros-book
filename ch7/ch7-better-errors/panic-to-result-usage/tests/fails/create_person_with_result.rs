use panic_to_result_macro::panic_to_result;

#[derive(Debug)]
pub struct Person {
    name: String,
    age: u32,
}

#[panic_to_result]
fn create_person_with_result(name: String, age: u32) -> Result<Person, String> {
    if age > 30 {
        panic!("I hope I die before I get old");
    }
    Ok(Person {
        name,
        age,
    })
}

fn main() {}
