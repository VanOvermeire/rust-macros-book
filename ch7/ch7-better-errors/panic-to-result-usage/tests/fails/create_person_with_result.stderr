error: this macro can only be applied to a function that does not return a Result. Signature: Result < Person, String >
  --> tests/fails/create_person_with_result.rs:10:1
   |
10 | fn create_person_with_result(name: String, age: u32) -> Result<Person, String> {
   | ^^
